addons/website_sale_loyalty/tests/test_shop_sale_coupon.py

```diff
@@ -586,3 +586,70 @@ def test_confirm_points_as_public_user(self):
         order.action_quotation_send()
         order.with_context(access_token=order.access_token, user=public_user).action_confirm()
         self.assertEqual(loyalty_card.points, 1)
+
+    def test_loyalty_card_archive_with_draft_points(self):
+        """
+        Verify that a loyalty card that has earned points from a draft sale order
+        cannot be archived, and that archived cards cannot be claimable in subsequent
+        sale orders.
+        """
+        website = self.env['website'].browse(1)
+        product = self.env['product.product'].create({
+            'name': 'Product A',
+            'list_price': 10,
+            'sale_ok': True,
+        })
+        loyalty_program = self.env['loyalty.program'].create({
+            'name': 'Loyalty Program',
+            'program_type': 'loyalty',
+            'trigger': 'auto',
+            'applies_on': 'both',
+            'rule_ids': [
+                Command.create({
+                    'reward_point_mode': 'unit',
+                    'reward_point_amount': 10,
+                    'product_ids': [product.id],
+                }),
+            ],
+            'reward_ids': [
+                Command.create({
+                    'reward_type': 'discount',
+                    'discount': 50,
+                    'discount_mode': 'percent',
+                    'discount_applicability': 'order',
+                    'required_points': 10,
+                }),
+            ],
+        })
+        loyalty_card = self.env['loyalty.card'].create({
+            'program_id': loyalty_program.id,
+            'partner_id': self.steve.id,
+            'points': 0,
+        })
+        sale_order = self.env['sale.order'].create({
+            'partner_id': self.steve.id,
+            'order_line': [
+                Command.create({
+                    'product_id': product.id
+                })
+            ],
+            'website_id': website.id,
+        })
+        sale_order._update_programs_and_rewards()
+        claimable_rewards = sale_order._get_claimable_rewards()
+        self.assertIsNotNone(claimable_rewards[loyalty_card])
```

Comment:
```suggestion
        self.assertTrue(claimable_rewards[loyalty_card])
```
It's possible that `_get_claimable_rewards` might get refactored to return dicts with `False` or empty recordsets or lists, so better just check whether the result is truthy rather than `not None`.